src/zcbenchmarks.cpp

```diff
@@ -88,3 +90,15 @@ double benchmark_create_joinsplit()
     assert(pourtx.Verify(*pzerocashParams));
     return ret;
 }
+
+double benchmark_solve_equihash()
+{
+    const char *testing = "testing";
```

Comment:
We may need more accuracy here, since `BasicSolve` doesn't necessarily succeed, and we want to measure for when it does. We should mine a block like normal:

```c++
CBlock pblock;
CEquihashInput I{pblock};
CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
ss << I;

Equihash eh {Params(CBaseChainParams::MAIN).EquihashN(), Params(CBaseChainParams::MAIN).EquihashK()};
crypto_generichash_blake2b_state eh_state;
eh.InitialiseState(eh_state);
crypto_generichash_blake2b_update(&eh_state, (unsigned char*)&ss[0], ss.size());

arith_uint256 nonce = 0;

while (true) {
    uint256 real_nonce = ArithToUint256(nonce);

    crypto_generichash_blake2b_state curr_state;
    curr_state = eh_state;
    crypto_generichash_blake2b_update(&curr_state,
                                      real_nonce.begin(),
                                      real_nonce.size());

    std::set<std::vector<unsigned int>> solns = eh.BasicSolve(curr_state);

    if (solns.size() > 0) {
        break;
    }

    nonce++;
}
```

... would be better, plus or minus whatever errors I left for you to fix. :)